### DISCUSSION TITLE
Resetting Jest / RTL state between tests

### DISCUSSION THREAD

**Discussion by TJHdev** [pmndrs/valtio#775] [Q&A]
Is there a straight forward way of resetting proxy state between RTL tests that can run in parallel?

Currently I have a simple store that I'd like to reset between each test. When using jotai it was simple as we can pass in a new Provider to the RTL render function but this isn't viable as I need to be able to use the valtio proxies outside of react.

`src/store/valtioStore.ts`

```
import { proxy } from 'valtio';

export const percentageCompleteProxy = proxy<{ percentageComplete: undefined | number }>({
  percentageComplete: undefined,
});

export const selectedFileProxy = proxy<{ selectedFile: File | { id: string } | undefined }>({
  selectedFile: undefined,
});
```

Any pointers or ideas would be much appreciated.

**Comment by dai-shi** [comment]
Have you looked at this? https://github.com/pmndrs/valtio/blob/734c24db74fbc6cf35d63b939f67b17602dbfe1e/docs/how-tos/how-to-reset-state.mdx#L5

**Reply by TJHdev** [reply]
Sorry, the question should have worded around isolating state between jest tests instead of resetting. 

I did a lot of attempts at isolating state between the Jest tests via dynamic imports. In the end I went back to a combination of Jotai and redux (our legacy global state).

**Reply by dai-shi** [reply]
That sounds tough but should be doable. For example, how would you do it if it were plain JavaScript objects.
